src/App.tsx

```diff
@@ -8,37 +8,87 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './api/api';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasPostsError, setHasPostsError] = useState(false);
+  const [hasUsersError, setHasUsersError] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => setHasUsersError(true))
+      .finally(() => setHasUsersError(false));
```

Comment:
So, it you catch an error - you set true
But, after that you set false, bc here is finally?